lib/pyfrc/test_support/pytest_plugin.py

```diff
@@ -59,6 +59,9 @@ def pytest_runtest_setup(self):
                        "If your robot class has an __init__ function, it must call super().__init__()!"
     
     def pytest_runtest_teardown(self, nextitem):
+        # Let any child threads run in realtime to allow cancelling if it
+        # has been implemented.
+        self._fake_time.freerun_children()
```

Comment:
I've tried to leave all the code in the wpilib library itself alone for this issue.
A timed call to join inside the pyfrc harness will always fail because it will wait here until the timeout, because the cancel() call on the TimerTask class is made later in the teardown process. If that cancel call was never made, the current builtin tests would pass for the example code.
I'm not sure what action we can take here in the test harness that is generic for all child threads.